catalog/gke/nodepools/default/node-iam.yaml

```diff
@@ -0,0 +1,61 @@
+# Service Account for GKE nodes
+apiVersion: iam.cnrm.cloud.google.com/v1beta1
+kind: IAMServiceAccount
+metadata:
+  name: gke-example-us-east4-default # {"$kpt-set":"krm-service-account-name"}
+  namespace: config-control # {"$kpt-set":"platform-namespace"}
+  annotations:
+    cnrm.cloud.google.com/project-id: platform-project-id # {"$kpt-set":"platform-project-id"}
+spec:
+  displayName: gke-example-us-east4-default # {"$kpt-set":"krm-service-account-name"}
+---
+# Allow fluentd to send logs to StackDriver
+apiVersion: iam.cnrm.cloud.google.com/v1beta1
+kind: IAMPolicyMember
+metadata:
+  name: logwriter-gke-example-us-east4-default # {"$kpt-set":"krm-logwriter-policy-name"}
+  namespace: config-control # {"$kpt-set":"platform-namespace"}
+spec:
+  memberFrom:
+    serviceAccountRef:
+      name: gke-example-us-east4-default # {"$kpt-set":"krm-service-account-name"}
+      namespace: config-control # {"$kpt-set":"platform-namespace"}
+  role: roles/logging.logWriter
+  resourceRef:
+    apiVersion: resourcemanager.cnrm.cloud.google.com/v1beta1
+    kind: Project
+    external: platform-project-id # {"$kpt-set":"platform-project-id"}
+---
+# Allow fluentd to send metrics to StackDriver
+apiVersion: iam.cnrm.cloud.google.com/v1beta1
+kind: IAMPolicyMember
+metadata:
+  name: metricwriter-gke-example-us-east4-default # {"$kpt-set":"krm-metricwriter-policy-name"}
+  namespace: config-control # {"$kpt-set":"platform-namespace"}
+spec:
+  memberFrom:
+    serviceAccountRef:
+      name: gke-example-us-east4-default # {"$kpt-set":"krm-service-account-name"}
+      namespace: config-control # {"$kpt-set":"platform-namespace"}
+  role: roles/monitoring.metricWriter
+  resourceRef:
+    apiVersion: resourcemanager.cnrm.cloud.google.com/v1beta1
+    kind: Project
+    external: platform-project-id # {"$kpt-set":"platform-project-id"}
+---
+# Allow kubelet/docker/containerd to read all artifacts/images in the platform-project-id project
```

Comment:
Considering the most of blueprint follows GKE best practices, should we be granting wide access to GAR in the platform project as opposed to having a separate config for managing GAR IAM at a more granular level?

Reply:
Does GAR even have granular permissions?

Reply:
@karlkfi yeah https://cloud.google.com/artifact-registry/docs/access-control#grant-repo
Not sure if in KCC but https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/artifact_registry_repository_iam

Reply:
There is no KCC resource for this, and I'm not sure we would want to use it anyway. Having to explicitly update cluster-level or nodepool-level IAM every time a new repository is added would be a big headache for tenants on multitenant clusters.

I think allow listing access to all images in the AR collocated in the same project with the cluster is totally reasonable and secure enough for most enterprise use cases.

Reply:
sgtm, let's also open a b/ to track resource FR internally.

Reply:
Done: https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/496